pkg/test/proxy/envoy/agent/agent.go

```diff
@@ -0,0 +1,368 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package agent
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	envoy_api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
```

Comment:
Please use templates - like the real agent. And if possible use the pkg/bootstrap and the template we use in real life.

Reply:
I'm purposely using objects because the number of listeners and clusters are dynamic.